backend/src/Altinn.Authorizationadmin/Altinn.AuthorizationAdmin.Core/Services/Implementation/DelegationsService.cs

```diff
@@ -45,16 +45,88 @@ public async Task<List<OfferedDelegations>> GetAllOfferedDelegations(int offered
+                {
+                    Delegation delegation = new Delegation();
+                    Party partyInfo = partyList.Find(p => p.PartyId == delegationChange.CoveredByPartyId);
+                    delegation.CoveredByName = partyInfo?.Name;
+                    delegation.CoveredByOrganizationNumber = Convert.ToInt32(partyInfo?.OrgNumber);
+                    delegation.CoveredByPartyId = delegationChange.CoveredByPartyId;
+                    delegation.OfferedByPartyId = delegationChange.OfferedByPartyId;
+                    delegation.PerformedByUserId = delegationChange.PerformedByUserId;
+                    delegation.Created = delegationChange.Created;
+                    resourceDelegation.Delegations.Add(delegation);
+                }
+
+                resourceDelegations.Add(resourceDelegation);
+            }
+
+            return resourceDelegations;
+        }
+
+        /// <inheritdoc/>
+        public async Task<List<ReceivedDelegation>> GetReceivedDelegationsAsync(int coveredByPartyId, ResourceType resourceType)
+        {
+            List<DelegationChange> delegations = await _delegationRepository.GetReceivedDelegationsAsync(coveredByPartyId, resourceType);
+            List<int> parties = new List<int>();
+            parties = delegations.Select(d => d.OfferedByPartyId).ToList();
+
+            List<ServiceResource> resources = new List<ServiceResource>();
+            List<string> resourceIds;
+            resourceIds = delegations.Select(d => d.ResourceId).ToList();
+            resources = await GetResources(resourceIds);
+
+            List<Party> partyList = await _partyClient.GetPartiesAsync(parties);
+            List<ReceivedDelegation> receivedDelegations = new List<ReceivedDelegation>();
+            foreach (Party party in partyList)
+            {
+                if (receivedDelegations.FindAll(rd => rd.OfferedByPartyId.Equals(party.PartyId)).Count <= 0)
+                {
+                    ReceivedDelegation receivedDelegation = new ReceivedDelegation();
+                    receivedDelegation.OfferedByPartyId = party.PartyId;
+                    receivedDelegation.ReporteeName = party.Name;
+                    List<DelegationChange> query = delegations.FindAll(d => d.CoveredByPartyId.Equals(coveredByPartyId) && d.OfferedByPartyId.Equals(party.PartyId));
+                    receivedDelegation.Resources = new List<ServiceResource>();
+
+                    foreach (DelegationChange delegation in query)
+                    {
+                        ServiceResource resource = resources.Find(d => d.Identifier == delegation.ResourceId);
+                        receivedDelegation.Resources.Add(resource);
+                    }
+
+                    receivedDelegations.Add(receivedDelegation);
+                }
+            }
+
+            return receivedDelegations;
+        }
+
+        private async Task<List<ServiceResource>> GetResources(List<string> resourceIds)
```

Comment:
GetResources could be moved to a ResourceRegistryHelper/Wrapper class as we will need both this and other integrations with the resource registry across multiple services in access management